tests/e2e/unified_test.go

```diff
@@ -153,6 +161,106 @@ func testFixturesInSeries(ctx context.Context, t *testing.T, testPause bool, can
 					createPausedCC(ctx, t, h.GetClient())
 				}
 
+				if os.Getenv("E2E_GCP_TARGET") == "vcr" {
+					// Stop recording after tests finish and write to cassette
+					t.Cleanup(func() {
+						err := h.VCRRecorder.Stop()
+						if err != nil {
+							t.Errorf("[VCR] Failed stop vcr recorder: %v", err)
+						}
+					})
+
+					replaceFunc := func(s string) string {
+						result := strings.Replace(s, uniqueID, "uniqueid111111", -1)
+						result = strings.Replace(result, project.ProjectID, "cnrm-user", -1)
+						return result
+					}
+
+					hook := func(i *cassette.Interaction) error {
+						var requestHeadersToRemove = []string{
+							"Authorization",
+							"User-Agent",
+						}
+						for _, header := range requestHeadersToRemove {
+							delete(i.Request.Headers, header)
+						}
+
+						var responseHeadersToRemove = []string{
+							"Cache-Control",
+							"Server",
+							"Vary",
+							"X-Content-Type-Options",
+							"X-Frame-Options",
+							"X-Xss-Protection",
+							"Date",
+							"Etag",
+						}
+						for _, header := range responseHeadersToRemove {
+							delete(i.Response.Headers, header)
+						}
+
+						i.Request.Body = replaceFunc(i.Request.Body)
+						i.Response.Body = replaceFunc(i.Response.Body)
+						i.Request.URL = replaceFunc(i.Request.URL)
+
+						s := i.Response.Body
+						obj := make(map[string]any)
+						if err := json.Unmarshal([]byte(s), &obj); err != nil {
+							klog.Fatalf("error from json.Unmarshal(%q): %v", s, err)
+						}
+						toReplace, _, _ := unstructured.NestedString(obj, "user")
+						if len(toReplace) != 0 {
+							s = strings.Replace(s, toReplace, "user@google.com", -1)
+						}
+
+						// A very hacky way to replace actual debug info from log.
+						if strings.Contains(s, "error") && strings.Contains(s, "debugInfo") {
+							keyword := "\"debugInfo\": \""
+							// Find index of the start of debugInfo string
+							startIndex := strings.Index(s, keyword)
+							// Find index of the end of debugInfo string, by searching for the end quote character
+							subString := s[startIndex+len(keyword):]
+							endIndex := strings.Index(subString, "\"")
+							// Use both indexes to get the string contains the message we need to remove
+							temp := s[startIndex+len(keyword) : startIndex+len(keyword)+endIndex]
+							// Replace
+							s = strings.Replace(s, temp, "fake debug info", -1)
```

Comment:
I think you can just remove it, IIUC it only shows up for (some) googlers / callers.  So for outside contributors, it might just be confusing.

Reply:
I plan to remove this this whole code block in #1393.

To speed up the test, I'd like to remove unnecessary interactions from the yaml file. For example, remove failed HTTP logs with error codes 404/400/403 and repeated logs that come from the retry transport.

I left a TODO here stating that the code will be removed in an upcoming PR.